src/components/MessageList/MessageList.js

```diff
@@ -39,664 +45,367 @@ const ErrorNotification = styled.View`
-              this.props.setFlatListRef && this.props.setFlatListRef(fl);
-            }}
-            data={messagesWithDates}
-            onScroll={this.handleScroll}
-            ListFooterComponent={HeaderComponent}
-            onEndReached={this.props.loadMore}
-            inverted
-            keyboardShouldPersistTaps='always'
-            keyExtractor={(item) =>
-              item.id ||
-              item.created_at ||
-              (item.date ? item.date.toISOString() : false) ||
-              uuidv4()
-            }
-            renderItem={({ item: message }) =>
-              this.renderItem(message, messageGroupStyles[message.id])
-            }
-            /** Disables the MessageList UI. Which means, message actions, reactions won't work. */
-            extraData={this.props.disabled}
-            maintainVisibleContentPosition={{
-              minIndexForVisible: 1,
-              autoscrollToTopThreshold: 10,
-            }}
-            {...this.props.additionalFlatListProps}
-          />
-          {this.props.TypingIndicator && showTypingIndicator && (
-            <TypingIndicatorContainer>
-              <TypingIndicator
-                typing={this.props.typing}
-                client={this.props.client}
-              />
-            </TypingIndicatorContainer>
-          )}
-          {this.state.newMessagesNotification && (
-            <MessageNotification
-              showNotification={this.state.newMessagesNotification}
-              onPress={this.goToNewMessages}
-            />
-          )}
-          {!this.state.online && (
-            <ErrorNotification>
-              <ErrorNotificationText>
-                {t('Connection failure, reconnecting now ...')}
-              </ErrorNotificationText>
-            </ErrorNotification>
-          )}
-        </View>
-      </React.Fragment>
+      <View style={{ flex: 1 }}>
+        <EmptyStateIndicator listType='message' />
+      </View>
     );
   }
-}
 
-export default withTranslationContext(withChannelContext(MessageList));
+  const messagesWithDates = insertDates(messages);
+  const messageGroupStyles = getGroupStyles(messagesWithDates, noGroupByUser);
+  const readData = getReadStates(messagesWithDates, read);
+
+  const messageList = messagesWithDates
+    .map((msg) => ({
+      ...msg,
+      readBy: readData[msg.id] || [],
+      groupStyles: messageGroupStyles[msg.id],
+    }))
+    .reverse();
+
+  return (
+    <>
+      {// Mask for edit state
+      editing && disableWhileEditing && (
+        <TouchableOpacity
+          style={{
+            position: 'absolute',
+            backgroundColor: 'black',
+            opacity: 0.4,
+            height: '100%',
+            width: '100%',
+            zIndex: 100,
+          }}
+          collapsable={false}
+          onPress={clearEditingState}
+        />
+      )}
+      <View
+        collapsable={false}
+        style={{ flex: 1, alignItems: 'center', width: '100%' }}
```

Comment:
width: 100% needed here with flex: 1?
```suggestion
        style={{ alignItems: 'center', flex: 1 }}
```

Reply:
flex is controlling the height here